src/effects/ladspa/LadspaEffect.cpp

```diff
@@ -1639,8 +1648,15 @@ LADSPA_Handle LadspaEffect::InitInstance(
    auto &controls = settings.controls;
    for (unsigned long p = 0; p < mData->PortCount; ++p) {
       LADSPA_PortDescriptor d = mData->PortDescriptors[p];
-      if (LADSPA_IS_PORT_CONTROL(d))
-         mData->connect_port(handle, p, &controls[p]);
+      if (LADSPA_IS_PORT_CONTROL(d)) {
+         if (LADSPA_IS_PORT_INPUT(d))
+            mData->connect_port(handle, p, &controls[p]);
+         else {
+            static LADSPA_Data sink;
+            mData->connect_port(handle, p,
+               pOutputs ? &pOutputs->controls[p] : &sink);
```

Comment:
Correct me if I'm wrong, but effect writes `LadspaEffectOutputs` on the processing thread, here you bind control directly to the location of these outputs, what prevents data races in that case?

Reply:
You are right.  I explained the bug I introduce knowingly here but then correct in the second PR.

The two should be merged as a whole -- or perhapsw the whole sequence of four should be -- but I thought the first PR was big enough already.